tests/phpunit/tests/rest-api/rest-attachments-controller.php

```diff
@@ -1923,6 +1923,37 @@ public function test_rest_insert_attachment_hooks_fire_once_on_create() {
 		$this->assertSame( 1, self::$rest_after_insert_attachment_count );
 	}
 
+	/**
+	 * Tests that the naming behavior of REST media uploads matches core media uploads.
+	 *
+	 * In particular, filenames with spaces should maintain the spaces rather than
```

Comment:
Should we test a `Filename.with.Punctuation.jpg`?

Reply:
From a usability perspective: `Filename.with.Punctuation.jpg` should work as well as `Filename with spaces.jpg` and `Filename with spaces - Also with some dashes - As topic separators.jpg`.

So would be great if the unit tests test for that too.

Reply:
Also filenames with file extensions which gets normalized in length or capizalization like `JPEG|jpeg|JPG` to `jpg` should work without issues with this. If you can guarantee this is 100% isolated, then no need to test it here. But I'm just giving some examples which we WILL experience "in the wild".

Reply:
Happy to expand the tests here, the main goal is that users should see the same behavior when uploading media in the block editor that they do int he media library.

Reply:
added